client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/steps/use-steps.tsx

```diff
@@ -45,33 +53,76 @@ interface StepsObject {
 
 const useStepsData = ( {
 	fromUrl,
+	lastCompleteStep,
 	migrationKey,
+	onComplete,
 	preparationError,
+	setCurrentStep,
+	setLastCompleteStep,
 	showMigrationKeyFallback,
 }: StepsDataOptions ): StepsData => {
 	const translate = useTranslate();
 
-	return [
+	const recordCompletedStepEvent = ( key ) => {
+		recordTracksEvent( 'calypso_site_migration_instructions_substep_complete', {
+			step: key,
+		} );
+	};
+
+	const onNextClick = ( index, key ) => {
+		setCurrentStep( index + 1 );
+
+		// When completing a step that wasn't completed yet.
+		if ( lastCompleteStep < index ) {
+			setLastCompleteStep( index );
+			recordCompletedStepEvent( key );
+		}
+	};
+
+	const onDoneClick = ( key ) => {
+		onComplete();
+		recordCompletedStepEvent( key );
+	};
+
+	const steps = [
```

Comment:
The original idea of the `useStepsData` was to be a kind of config file to be very simple. I changed it to a hook only to use the `useTranslate`.

That said, I'd like to give you a suggestion here to see what you think.

Instead of having all the steps receiving the events, and creating the same structure of buttons, WDYT of just adding the action label to the `useStepsData`, and create a new component for the buttons and using it in the `useSteps`? If you like the idea, it could be in the return of the `useSteps`, just adding the buttons to the content, like this:

```js
{
    ...
    expandable: {
        ...
        content: (
            <>
                {step.content}
                <StepCTA {...} label={ step.CTALabel } onCompleteStep={ ... } />
            </>
        ),
        ...
    }
    ...
}
```

I think this could help simplify the props/arguments we're sending around.